extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerRecorder.java

```diff
@@ -70,4 +95,49 @@
     public void setupVertxTracer() {
         vertxTracingAdapter.init();
     }
+
+    /* RUNTIME INIT */
+    public void setupResources(TracerRuntimeConfig config) {
+        // Find all Resource instances
+        Instance<Resource> allResources = CDI.current()
```

Comment:
This can be moved above to static init, so runtime init is only the config properties

Reply:
Unfortunately I think we actually can't, resources are fetching system environment values, if we move that to static init AWS resources won't work, also OS, Host and other resources wouldn't work properly

Reply:
Nuts, you're right.

Ok, we leave it as is then